linetools/lists/linelist.py

```diff
@@ -344,7 +349,195 @@ def all_transitions(self,line):
+
+        """    
+
+        #Check correct format
+        if isinstance(wvlims, tuple): # tuple
+            if all(isinstance(wvlim,Quantity) for wvlim in wvlims):
+                pass
+            else:
+                raise ValueError('Elements of wvlims have to be of class Quantity; correct format please')
+        elif isinstance(wvlims, Quantity): # or quantity
+            pass
+        else:
+            raise ValueError('wvlims has to be tuple or Quantity')
+        if len(wvlims) != 2:
+            raise SyntaxError('wlims has to be of size len()== 2; Please correct format')
+        if wvlims[0] >= wvlims[1]:
+            raise ValueError('Minimum limit (wlims[0]) is not smaller than maximum limit (wlims[1]); please correct')
+
+        if n_max < 1:
+            return None
+
+        data = self.all_transitions(line)
+        wv_obs = data['wrest'] * (1 + z)
+        # condition to be within wvrange
+        cond = (wv_obs >= wvlims[0]) & (wv_obs <= wvlims[1])
+        if np.sum(cond) == 0:
+            if verbose:
+                print('[strongest_transitions] Warning: no transitions found within wvlims; returning None')
+            return None
+        elif isinstance(data,dict): #Only 1 case from a dict format
+            return data
+        elif np.sum(cond) == 1: #only 1 case from a QTable format
+            name = data[cond]['name'][0]
+            return self.__getitem__(name)
+        else:
+            #remove transitions out of range
+            data = data[cond]
+            #add strength column to data
+            strength = data['wrest'] * data['f']
+            data.add_column(Column(data=strength,name='strength'))
+            data.sort('strength')
+            data.reverse()
+            #keep only the first n_max or less
+            data = data[:n_max]
+            #remove column strength for consistency
+            data.remove_column('strength')
+            if len(data) == 1: #return a dictionary
+                name = data['name'][0]
+                return self.__getitem__(name)
+            else:
+                return data
+
+    def available_transitions(self, wvlims, z, n_max=100,n_max_tuple=3, min_strength=4):
+        """For a given wavelength range in Angstroms, wvlims=(wv_min,wv_max),
+        this function retrieves the n_max_tuple strongest transitions per each ion 
+        species in the LineList available at redshift z within such a wavelength 
+        range. The output is sorted by strength of the strongest available transition 
+        per ion species, modualted by abundance.
+
+        Parameters:
+        ----------
+        wvlims : tuple of Quantity
+            Wavelength range, e.g. wvlims=(1100*u.AA, 3200*u.AA)
+        z : float
+            Redshift
+        n_max : int, optional
+            Maximum number of transitions retrieved
+        n_max_tuple : int, optional
```

Comment:
The idea is to focus in the strongest available transition for a given ion species, and then look up to n_max_tuple transitions of that same species independently of its strength compared to others. This is helpful for the line identification process, where finding evidence of doublets (or tuplets) adds considerably diagnostic power. We can, of course, discuss further what these functions should really do.